flink/src/main/java/org/apache/iceberg/flink/IcebergStreamWriter.java

```diff
@@ -0,0 +1,226 @@
+    Row value = element.getValue();
+    writer.append(value);
+
+    // Emit the data file entries to downstream committer operator if there exist any complete files.
+    writer.pollCompleteFiles().forEach(this::emit);
+  }
+
+  @Override
+  public void dispose() throws Exception {
+    super.dispose();
+    if (writer != null) {
+      writer.close();
+      writer = null;
+    }
+  }
+
+  @Override
+  public void endInput() throws IOException {
+    // For bounded stream, it may don't enable the checkpoint mechanism so we'd better to emit the remaining
+    // data files to downstream before closing the writer so that we won't miss any of them.
+    writer.close();
+    for (DataFile dataFile : writer.pollCompleteFiles()) {
+      emit(dataFile);
+    }
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this)
+        .add("table_path", tablePath)
+        .add("subtask_id", subTaskId)
+        .add("attempt_id", attemptId)
+        .toString();
+  }
+
+  private void emit(DataFile dataFile) {
+    output.collect(new StreamRecord<>(dataFile));
+  }
+
+  private FileFormat getFileFormat() {
+    String formatString = properties.getOrDefault(DEFAULT_FILE_FORMAT, DEFAULT_FILE_FORMAT_DEFAULT);
+    return FileFormat.valueOf(formatString.toUpperCase(Locale.ENGLISH));
+  }
+
+  private long getTargetFileSizeBytes() {
+    return PropertyUtil.propertyAsLong(properties,
+        WRITE_TARGET_FILE_SIZE_BYTES,
+        WRITE_TARGET_FILE_SIZE_BYTES_DEFAULT);
+  }
+
+  private static class FlinkFileAppenderFactory implements FileAppenderFactory<Row> {
+    private final Schema schema;
+    private final Map<String, String> props;
+
+    private FlinkFileAppenderFactory(Schema schema, Map<String, String> props) {
+      this.schema = schema;
+      this.props = props;
+    }
+
+    @Override
+    public FileAppender<Row> newAppender(OutputFile outputFile, FileFormat format) {
+      MetricsConfig metricsConfig = MetricsConfig.fromProperties(props);
+      try {
+        switch (format) {
+          case PARQUET:
+            return Parquet.write(outputFile)
+                .createWriterFunc(FlinkParquetWriters::buildWriter)
+                .setAll(props)
+                .metricsConfig(metricsConfig)
+                .schema(schema)
+                .overwrite()
+                .build();
+
+          case AVRO:
+            return Avro.write(outputFile)
+                .createWriterFunc(FlinkAvroWriter::new)
+                .setAll(props)
+                .schema(schema)
+                .overwrite()
+                .build();
+
+          case ORC:
+          default:
+            throw new UnsupportedOperationException("Cannot write unknown format: " + format);
+        }
+      } catch (IOException e) {
+        throw new UncheckedIOException(e);
+      }
+    }
+  }
+
+  static IcebergStreamWriter createStreamWriter(String path, TableSchema tableSchema, Configuration conf) {
```

Comment:
Can we just pass a Iceberg `Table` instead of `path` here? We can get ride of `TableUtil.findTable` (Or just move it to test).